src/config/defaults_ranking.go

```diff
@@ -27,3 +31,33 @@ func EmptyRanking(engs []engines.Name) CategoryRanking {
 
 	return rnk
 }
+
+func ReqPrefOthRanking(req []engines.Name, pref []engines.Name, oth []engines.Name) CategoryRanking {
```

Comment:
Make results have higher score in the following order:
1) Required / RequiredByOrigin
2) Preferred / PreferredByOrigin
3) Others